### DISCUSSION TITLE
Any way of utilizing Perl compatible regular expressions?

### DISCUSSION THREAD

**Discussion by fade2gray** [dnGrep/dnGrep#683] [Q&A] [answered]
I notice that the `\Q` and `\E` literal string tokens, mentioned in the [Wiki](https://github.com/dnGrep/dnGrep/wiki/Regular-Expressions), don't/no longer work, also the universal newline token `\R` is unavailable.

Note: the backtick character `` ` `` has been used instead of the pipe character `|` in the wiki.

**Comment by doug24** [comment, marked as answer]
dnGrep uses the .net Regular Expression library, so we are limited to that syntax.  I have (some time ago) looked for another library to offer as an alternative but did not find one that seemed reasonable to integrate in a .net application.

You are right the \Q...\E escape sequence does not work in the .net regular expressions.  The wiki needs an update.  You can escape individual tokens in a .net regular expression like this:  `\*\+` to search for the string literal `*+`

Newlines.  I don't think there is a \R in .net regular expressions either.  You can use \r and \n.   But in .net regular expressions the `$` end of line token will only match on \n.   Which is a big problem on Windows where most text files use \r\n, and most users are not aware of this and won't know how to construct a regex to match on \r\n.   So dnGrep contains a bunch of code to convert \r\n to \n before doing a regex match, and then converting them back.  I'd suggest trying your search without trying to accommodate both/either line ending and see if dnGrep handles it for you.
